demos/defender/aws_iot_demo_defender.c

```diff
@@ -268,37 +274,51 @@ int RunDefenderDemo( bool awsIotMqttMode,
     if( status == EXIT_SUCCESS )
     {
         /* Create MQTT Connection */
-        mqttStatus = _establishMqttConnection( pIdentifier,
-                                               pNetworkServerInfo,
-                                               pNetworkCredentialInfo,
-                                               pNetworkInterface,
-                                               &mqttConnection );
+        status = _establishMqttConnection( pIdentifier,
+                                           pNetworkServerInfo,
+                                           pNetworkCredentialInfo,
+                                           pNetworkInterface,
+                                           &mqttConnection );
 
-        if( mqttStatus != IOT_MQTT_SUCCESS )
+        if( status != EXIT_SUCCESS )
         {
-            IotLogError( "Failed to Create MQTT Connection, error: %s", IotMqtt_strerror( mqttStatus ) );
+            IotLogError( "Failed to create the MQTT Connection." );
             IotMqtt_Cleanup();
             defenderResult = AWS_IOT_DEFENDER_INTERNAL_FAILURE;
         }
-        else
-        {
-            /* Initialize start info and call defender Start API */
-            startInfo.pClientIdentifier = pIdentifier;
-            startInfo.clientIdentifierLength = ( uint16_t ) strlen( pIdentifier );
-            startInfo.callback = callback;
-            startInfo.mqttConnection = mqttConnection;
-            defenderResult = AwsIotDefender_Start( &startInfo );
-        }
+    }
+
+    if( status == EXIT_SUCCESS )
+    {
+        /* Initialize start info and call defender Start API */
+        startInfo.pClientIdentifier = pIdentifier;
+        startInfo.clientIdentifierLength = ( uint16_t ) strlen( pIdentifier );
+        startInfo.callback = callback;
+        startInfo.mqttConnection = mqttConnection;
+        defenderResult = AwsIotDefender_Start( &startInfo );
 
         if( defenderResult == AWS_IOT_DEFENDER_SUCCESS )
         {
-            /* Let it run for 3 seconds */
+            /* Let the Device Defender library run for 3 seconds before stopping.
+             * This is to allow enough time for the AWS IoT Device Defender
+             * service to accept the metrics. When the metrics are accepted, the
+             * application is notified in _defenderCallback() with an event type
+             * of AWS_IOT_DEFENDER_METRICS_ACCEPTED. Also, upon metrics acceptance,
+             * the Defender library will print "Metrics report was accepted by
+             * defender service." and the variable metricsAccepted, set in the
+             * callback context, will be set to true. */
```

Comment:
```suggestion
            /* Let the Device Defender Library run for 3 seconds before stopping.
             * This is to allow enough time for the AWS IoT Device Defender
             * Service to accept the metrics report.
             * 
             * The following happens when the metrics report is accepted by the AWS IoT
             * Device Defender Service:
             *
             * 1. The application is notified in _defenderCallback() with an event
             *    type of AWS_IOT_DEFENDER_METRICS_ACCEPTED. In this demo, the
             *    callback sets the variable metricsAccepted to true which is passed
             *    as the callback context.
             * 2. The Defender library prints "Metrics report was accepted by
             *     defender service."
             *
             * It is okay to pass the local variable metricsAccepted in the callback
             * context because the Device Defender Library is stopped in this function
             * itself. Therefore, the callback can never execute after this function
             * has exited.
             */
```

Reply:
Thank you.